pyiron_gui/project/project_browser.py

```diff
@@ -23,6 +29,207 @@
+
+        def click_button(b):
+            self.output.clear_output()
+            self.display(obj=obj)
+
+        if isinstance(obj, PyironWrapper) and obj.has_self_representation:
+            button = widgets.Button(description="Re-plot " + obj.name)
+            button.on_click(click_button)
+            self.buttons.children = tuple([button])
+
+        self.refresh()
+
+    def display(self, obj, default_output=None):
+        self._update_buttons(obj)
+        with self.output:
+            if obj is None and default_output is None:
+                raise TypeError("Given 'obj' is of 'NoneType'.")
+            elif obj is None:
+                print(default_output)
+            elif isinstance(obj, PyironWrapper):
+                plt.ioff()
+                to_display = obj.self_representation()
+                if to_display is not None:
+                    display(obj.self_representation())
+            else:
+                plt.ioff()
+                display(self._output_conv(obj))
+
+    def _output_conv(self, obj):
+        if hasattr(obj, '_repr_html_'):
+            return obj  # ._repr_html_()
+        eol = os.linesep
+        if self._debug:
+            print('node: ', type(obj))
+        if isinstance(obj, str):
+            return (obj)
+        elif isinstance(obj, nbformat.notebooknode.NotebookNode):
+            html_exporter = nbconvert.HTMLExporter()
+            #html_exporter.template_name = "basic"
+            html_exporter.template_name = "classic"
+            (html_output, resources) = html_exporter.from_notebook_node(obj)
+            return HTML(html_output)
+        elif isinstance(obj, dict):
+            dic = {'': list(obj.keys()), ' ': list(obj.values())}
+            return pandas.DataFrame(dic)
+        elif isinstance(obj, (int, float)):
+            return str(obj)
+        elif isinstance(obj, list):
+            max_length = 2000  # performance of widget above is extremely poor
+            if len(obj) < max_length:
+                return str(''.join(obj))
+            else:
+                return str(''.join(obj[:max_length]) +
+                           eol + ' .... file too long: skipped ....')
+        elif isinstance(obj, np.ndarray):
+            return self.plot_array(obj)
+        elif str(type(obj)).split('.')[0] == "<class 'PIL":
+            try:
+                data_cp = obj.copy()
+                data_cp.thumbnail((800, 800))
+            except:
+                data_cp = obj
+            return data_cp
+        else:
+            return obj
+
+    def plot_array(self, val):
+        if self.fig is None:
+            self.fig, self.ax = plt.subplots()
+        else:
+            self.ax.clear()
+
+        if val.ndim == 1:
+            self.ax.plot(val)
+        elif val.ndim == 2:
+            if len(val) == 1:
+                self.ax.plot(val[0])
+            else:
+                self.ax.plot(val)
+        elif val.ndim == 3:
+            self.ax.plot(val[:, :, 0])
+
```

Comment:
```suggestion
else:
    raise ValueError(f'Plotting implemented for 1-, 2-, and 3-dimensions, but got dim = {val.ndim}')
```

Reply:
I a non-GUI framework I would like your suggestion, however, in the GUI this always looks ugly to raise errors on a click on an unknown field...
I will print your message and return the np.array as is? Or just print something like  `f"This is a numpy array of dim = {val.ndim}, however, plotting is only implemented for up to 3-dimensions"`

Reply:
Yeah, either is fine. As long as the invalid use case doesn't pass silently, I'm happy.